action.yml

```diff
@@ -0,0 +1,129 @@
+name: 'Build and push images to registry'
+description: 'This GitHub Action builds a container image and pushes it to the specified registry.'
+author: 'RHSCL team'
+branding:
+  icon: circle
+  color: blue
+
+inputs:
+  registry:
+    description: 'Registry to push container image to'
+    required: true
+  registry_namespace:
+    description: 'Namespace of the registry, where the image would be pushed'
+    required: true
+  registry_username:
+    description: 'Login to specified registry'
+    required: true
+  registry_token:
+    description: 'Token to access the specified registry'
+    required: true
+  version:
+    description: 'Version of the built image'
+    required: true
+  tag:
+    description: 'Tag of the built image'
+    required: true
+  dockerfile:
+    description: 'Dockerfile to build the image'
+    required: false
+    default: 'Dockerfile'
+  docker_context:
+    description: 'Docker build context'
+    required: false
+    default: '.'
+  use_distgen:
+    description: 'The action will use distgen for generating dockerfiles if true'
+    required: false
+    default: 'false'
+
+runs:
+  using: "composite"
+  steps:
+    - name: Check if Action is run from sclorg repository
+      if: github.repository_owner != 'sclorg'
+      shell: bash
+      run: |
+        echo "This action can be run only for sclorg repositories"
+        exit 1
+
+    - name: Checkout git
+      uses: actions/checkout@v2
+      with:
+       submodules: true
```

Comment:
there is missing
'          fetch-depth: 0'

Reply:
Do we really need this? I did not find a use case, where all history for all branches and tags had to be fetched.
```
# Number of commits to fetch. 0 indicates all history for all branches and tags.
# Default: 1
fetch-depth: ''
```

Reply:
I see. let's leave it as default.